appholder/src/main/java/com/android/identity/wallet/MainActivity.kt

```diff
@@ -47,6 +64,31 @@ class MainActivity : AppCompatActivity() {
         setupNfc()
         onNewIntent(intent)
         Logger.setDebugEnabled(PreferencesHelper.isDebugLoggingEnabled())
+
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
+        if (ActivityCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+
+            requestPermissions(
+                listOf(
+                    Manifest.permission.ACCESS_FINE_LOCATION
+                ).toTypedArray(),
+                LOCATION_REQUEST_CODE
+            )
+        }
+
+        fusedLocationClient.lastLocation.addOnSuccessListener { lastKnownLocation ->
+            lastKnownLocation?.let {
```

Comment:
once we have a last known location we record the lat/long in the presentationLogStore